pkg/utils/metrics.go

```diff
@@ -45,21 +46,126 @@ func NewAgentMetrics(ai *AgentInfo) AgentMetrics {
 		Help:        "Total number of reports (keepalive messages) from the agent.",
 	})
 
-	if counter, ok := tryRegister(am.ErrorCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ErrorCount); !ok {
 		// use existing counter
 		am.ErrorCount = counter
 	}
-	if counter, ok := tryRegister(am.ReportCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ReportCount); !ok {
 		// use existing counter
 		am.ReportCount = counter
 	}
 
+  am.ProbeConnectionResult = prometheus.NewGaugeVec(
+    prometheus.GaugeOpts{
+      Namespace:   "ncagent",
+      Name:        "http_probe_connection_result",
+      Help:        "Connection result: 0 - error, 1 - success",
+    },
+    []string{"agent", "url"},
```

Comment:
Please see UpdateAgentProbeMetrics() function. We use those labels in prometheus.